Night1 just beer

NickoLefrench opened the pull request:
Structure for Patrons to order only Beer on night one is functional, but probably not logical.

Some structure for Cocktails to be the onky order on night 2 is also in place, but commented out

The over abundance of comments is for me to remember what I'm doing

NickoLefrench committed:
Start Progression

Created Progression Manager
Allows for adjustable configurations for each night
not working, need to add info to BarPatron and Cocktail shit

NickoLefrench committed:
Setting up Progression, BeerTap and PatronManager

Adjustments to BeerTap to make singleton
ProgressionManager adjustments
BarPatron and PatronManager adjusted to work in tandem with Progressmanager
shit doesnt fucking work yet fuick hohsdofipu;hbdaf

NickoLefrench committed:
code is functional

Structure is set up so batrons SHOULD only order beer on the 1st night
1 still orders cocktail

NickoLefrench commented:
I think the logic of setting allowed orders is only applied to the first patron spawned....will look more into it

vanyamil commented on Assets/Scripts/Characters/BarPatron.cs:
```diff
@@ -41,6 +45,30 @@ public enum State
+        //what is the batron allowed ordering
+        public enum OrderType { Beer, Cocktail }
+
+        private List<OrderType> allowedOrders;
```
Consider replacing this by a List<int> (indexes in OrderOptions/OrderSprites that are legal), or List<bool> (for each item in those arrays, true/false is allowed to order). At the very least, should not make a new Enum here - use existing Item.ItemType.

NickoLefrench replied:
done, and functional, now patrons just walk away once they hit their seat?

NickoLefrench replied:
I think Setting Orders will have to be done in an awake....will look into it on sunday

NickoLefrench replied:
adjusted, using an item type list for all the order options GetAllowedOrders, DecideOrder etc. Patrons sit and order just beer on first night

vanyamil commented on Assets/Scripts/Characters/BarPatron.cs:
```diff
@@ -41,6 +45,30 @@ public enum State
+            Debug.Log($"Patron {name}: Allowed orders updated to {string.Join(", ", orders)}");
+        }
+
+        public OrderType GetOrder()
```
Delete this and RegenerateOrder. Instead, use the allowedOrders logic inside of SelectOrderParameters.

NickoLefrench replied:
Regenerate More was more of an attempt to ensure Bar Patrons fetch the correct List of Allowed Ordders, but obviously it didnt work so i will remove it

will focus on SelectOrderParameters

NickoLefrench replied:
removed regenerate order

vanyamil commented on Assets/Scripts/Characters/BarPatron.cs:
```diff
@@ -214,5 +242,11 @@ public override void Interact(PlayerInteraction player)
             }
         }
+
+        public void RegenerateOrder()
```
As above, can delete

vanyamil commented on Assets/Scripts/Interactables/BeerTap.cs:
```diff
@@ -23,6 +23,19 @@ public class BeerTap : InteractableObject
         private bool detectedInputDuringMiniGame = false;
 
+        //setting up BeerTap as singleton
+        public static BeerTap Instance { get; private set; }
```
Try to remove logic if possible. Our plan of disabling whole BeerTap game object in fact cannot work (because you wouldn't be able to find the game object if it's disabled). How about in `private void Start`, doing a FindObjectsOfType<BeerTap>, and storing that array? This would allow for multiple BeerTaps.

NickoLefrench replied:
that makes sense, but does it matter if we are having one set level across the nights? or is it more in the sense of structure for further expansion

vanyamil replied:
Just better code

NickoLefrench replied:
Removed singleton.  added a static AllBeerTaps list to track all active BeerTap instances in awake, ProgressionManager now iterates over all BeerTap instances and toggles their interactability based on the current night configuration

vanyamil commented on Assets/Scripts/Interactables/BeerTap.cs:
```diff
@@ -128,5 +141,16 @@ private IEnumerator AnimateMiniGame(bool movingRight)
             MiniGameUIParent.SetActive(false);
         }
+
+        public void SetInteractable(bool isInteractable)
```
Can also hide renderer?

NickoLefrench replied:
I dsont understand the question, you want us to just hide the beertap completly?

vanyamil replied:
I thought you said we should make it invisible too, maybe am wrong

NickoLefrench replied:
Crud did I say that? I mean we can make it invisible when its not needed, I just thought it would be simpler to disable its interactability.

NickoLefrench replied:
I dont think its necessarily required to disable it or make it invisble for now. It can be polished later. Or jset to have 3 different levels#sizes for what is available for each night

vanyamil commented on Assets/Scripts/Managers/PatronManager.cs:
```diff
@@ -143,6 +143,50 @@ private void SpawnNewPatron()
+            newPatron.SetAllowedOrders(allowedOrders);
+        }
+
+        public List<BarPatron> GetActivePatrons()
```
Is this function useful anywhere else inside this file?

NickoLefrench replied:
Yeah its called in SetPatronOrderPreferences to set the order for each patron who found a seat

vanyamil commented on Assets/Scripts/Managers/PatronManager.cs:
```diff
@@ -143,6 +143,50 @@ private void SpawnNewPatron()
+            foreach (var patron in activePatrons)
+            {
+                patron.SetAllowedOrders(allowedOrders);
+                patron.RegenerateOrder();
```
No need to regenerate.

vanyamil commented on Assets/Scripts/Managers/PatronManager.cs:
```diff
@@ -143,6 +143,50 @@ private void SpawnNewPatron()
+        {
+            int currentNight = ProgressionManager.Instance.GetCurrentNight();
+
+            return currentNight switch
```
I don't like this formatting, I'm surprised it's legal, but it's cool to see!

NickoLefrench replied:
It just works -Todd Howard

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,147 @@
+    {
+        public bool enableBeerTap;                           //For BeerTap
+        public bool enableCocktailTools;                    //for CocktailMiniGame and CocktailFridge
+        public List<BarPatron.OrderType> allowedOrders;    //for BarPatron
```
As above, remove new enum

NickoLefrench replied:
it just works - Todd Howard

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,147 @@
+        currentNight = night;
+
+        // Apply night-specific configuration
+        NightConfig config = nightConfigs[currentNight - 1];
```
Dangerous! What if you're on night 4? What if no configs set up?

NickoLefrench replied:
By night 3 it should set to the default of everything avaliable. As in configs should not have an affect by night 3, unless we want to intergrate  or tweak the speed o patrons or something

vanyamil replied:
This should still be better guarded. If this fn is public, anyone can call with any value

vanyamil replied:
@NickoLefrench still needs a safety.

vanyamil replied:
Actually, this should now be from GetAllowedOrders, not from nightConfigs? need cleanup

NickoLefrench replied:
Done, i made a guard for when night < 1, i check if nightConfigs has enough entries, but i had to Make GetAllowedOrders() in PatronManager Public. I think its good enough

NickoLefrench replied:
@vanyamil

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,147 @@
+    private void UpdateInteractables(NightConfig config)
+    {
+        //enabling beer interaction, enable/disable in Beetap
+        if (BeerTap.Instance != null)
```
As above, replace by an array of all BeerTaps

NickoLefrench replied:
done

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,147 @@
+        }
+    }
+
+    public void SetPatronOrderPreferences(List<ItemType> orders)
```
Naming is confusing. Function above calls this exact function name, but in PatronManager? Why isn't it in here instead? Difficult to reason about code.

NickoLefrench replied:
Soo i renamed the SetPatronOrderPreferences in ProgressionManager to SetPatronOrderProgression to clarify things....but it appeared it was never called in the first place?

NickoLefrench replied:
I removed it, was a duplicate in ProgressionManager

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,147 @@
+        {
+            // Convert the orders to the expected type
+            List<BarPatron.OrderType> allowedOrders = orders
+                .Cast<BarPatron.OrderType>()
```
This cast is dangerous. As above, removing new enum will affect this.

NickoLefrench replied:
removing the enum and replacing it as a list as above removed the dangerous casting

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,147 @@
+    }
+
+    //config for night 1, will probably change with cocktail implementation
+    NightConfig night1Config = new NightConfig
```
Not actually used! Ideally use config inside of editor, and delete this.

vanyamil reviewed (changes requested):
Good work, nitpicks all over.

NickoLefrench committed:
small changes

NickoLefrench committed:
replace enum with list OrderType

I am sysiphus fighting the hydra

NickoLefrench committed:
removed bugs

So its functional with Item.Item type as lists referenced for progression on drinkgs, but now patrons immediatly walk away when sitting down

NickoLefrench committed:
NIghtconfigurations done in inspector

Can see allowed orders in console, patrons immediatly walk away

NickoLefrench committed:
played with nightconfig in inspector

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    //config for night 1, will probably change with cocktail implementation
+   /* commenting out for now to se if adjusted it to be adjustable in the inspactor
```
don't forget to delete

NickoLefrench replied:
will do

NickoLefrench committed:
Removed unecessary comments

removed parts of code that was commented out for cleanliness

NickoLefrench committed:
Changed Beertap singleton to instance in awake

 ProgressionManager now iterates over all BeerTap instances and toggles their interactability based on the current night configuration

vanyamil commented on Assets/Scripts/Interactables/BeerTap.cs:
```diff
@@ -23,17 +24,26 @@ public class BeerTap : InteractableObject
+        {
+            if (AllBeerTaps.Contains(this))
+            {
+                AllBeerTaps.Remove(this);
```
Generally good, but check the docs of .Remove - I think it runs a Contains implicitely. It's not like it can delete some other instance... right?

vanyamil commented on Assets/Scripts/Managers/ProgressionManager.cs:
```diff
@@ -89,17 +89,12 @@ public void StartNight(int night)
-        //enabling beer interaction, enable/disable in Beetap
-        if (BeerTap.Instance != null)
+        // Loop through all active BeerTap instances
+        foreach (var beerTap in BeerTap.AllBeerTaps)
```
Can the var be replaced by a type?

NickoLefrench committed:
debugging orders

reshresher on project state
debugging on order paramaters

NickoLefrench committed:
BarPatrons order only Beer

adjusted night duration to 30seconds
Customers order just beer for now

NickoLefrench committed:
Adding safety for nightconfig

safety added for night config so out of bounds exception wont occur
Set GetAllwoedOrders to public to the night config is set from that nights progression

NickoLefrench committed:
added updated gdd to project files
